### PR TITLE
strawberryperl: conan v2

### PR DESCRIPTION
**Pull request by prince-chrismc**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by prince-chrismc** [comment]
retriggering v2 pipeline since I missed the logs

**Comment by prince-chrismc** [comment]
For sanity, conan v1 can still build openssl 

image

**Comment by prince-chrismc** [comment]
🎉 

image

File: recipes/strawberryperl/all/test_package/conanfile.py [hunk 1]
```diff
@@ -1,12 +1,20 @@
 import os
-from conans import ConanFile, tools
+from conan import ConanFile
+from conan.tools.build import can_run
+from conan.tools.env import VirtualBuildEnv
+from conan.tools.scm import Version
```
Comment (uilianries) [inline recipes/strawberryperl/all/test_package/conanfile.py:5]:
```suggestion
```

Not used.

**Review by uilianries — APPROVED** [review]
LGTM

**Review by jwillikers — APPROVED** [review]

File: recipes/strawberryperl/all/conanfile.py [hunk 2]
```diff
@@ -1,39 +1,54 @@
...
-        self.output.info("Appending PATH environment variable: %s" % bin_path)
-        self.env_info.PATH.append(bin_path)
+        # TODO remove once conan v2 is the only support and recipes have been migrated
+        if Version(conan_version) < "2.0.0-beta":
+            bin_path = os.path.join(self.package_folder, "bin")
+            self.env_info.PATH.append(bin_path)
 
-        self.user_info.perl = os.path.join(self.package_folder, "bin", "perl.exe").replace("\\", "/")
+        perl_path = os.path.join(self.package_folder, "bin", "perl.exe").replace("\\", "/")
+        self.conf_info.define("user.strawberryperl:perl", perl_path)
```
Comment (SpaceIm) [inline recipes/strawberryperl/all/conanfile.py:52]: Doesn't exist in conf list, is it allowed or even intended in conan philosophy?

File: recipes/strawberryperl/all/test_package/conanfile.py [hunk 3]
```diff
@@ -1,12 +1,20 @@
...
+        self.tool_requires(self.tested_reference_str)
 
     def test(self):
-        if not tools.cross_building(self):
-            self.run("perl --version", run_environment=True)
+        if can_run(self):
+            self.run("perl --version")
             perl_script = os.path.join(self.source_folder, "list_files.pl")
-            self.run("perl {}".format(perl_script), run_environment=True)
+            self.run(f"perl {perl_script}", env="conanbuild")
```
Comment (SpaceIm) [inline recipes/strawberryperl/all/test_package/conanfile.py:20]: conanbuild is the default value of env argument of self.run

File: recipes/strawberryperl/all/conanfile.py (hunk 2 above)
Comment (prince-chrismc) [reply recipes/strawberryperl/all/conanfile.py:52]: #13490 :)

File: recipes/strawberryperl/all/test_package/conanfile.py (hunk 3 above)
Comment (prince-chrismc) [reply recipes/strawberryperl/all/test_package/conanfile.py:20]: Both are tested

Comment (SpaceIm) [reply recipes/strawberryperl/all/test_package/conanfile.py:20]: both what?

File: recipes/strawberryperl/all/conanfile.py (hunk 2 above)
Comment (SpaceIm) [reply recipes/strawberryperl/all/conanfile.py:52]: thanks

**Pull request merged**